bundles/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpBinding.java

```diff
@@ -303,8 +308,31 @@ private void formatAndExecute(String itemName, Command command, Type value) {
             url = String.format(url, Calendar.getInstance().getTime(), value);
         }
 
-        if (isNotBlank(httpMethod) && isNotBlank(url)) {
-            HttpUtil.executeUrl(httpMethod, url, provider.getHttpHeaders(itemName, command), null, null, timeout);
+        String body = provider.getBody(itemName, command);
+        InputStream stream = null;
+
+        if (StringUtils.isNotBlank(httpMethod) && StringUtils.isNotBlank(url)) {
+            if (httpMethod.equals("POST") && StringUtils.isNotBlank(body)) {
+                try {
+                    stream = IOUtils.toInputStream(body, "UTF-8");
+                } catch (IOException ioe) {
+                    logger.error("Failed to convert the specified body into an acceptable input stream.", ioe);
+                    logger.error("Body contents: {}", body);
+                }
+                logger.debug("Executing url '{}' via method {}, with body content '{}'", url, httpMethod, body);
+                HttpUtil.executeUrl(httpMethod, url, provider.getHttpHeaders(itemName, command), stream, "text/plain",
```

Comment:
I would think so, but there are probably some deficient apps out there that insist on having a different content-type in the header.

There doesn't seem to be a very good way of allowing the end user to specify this in a binding, though. I considered adding new POST commands that would carry different content-types (eg. POSTJ for json, POSTX for xml).  But that seems like an ugly bit of hackery.